### PR TITLE
implemented 'prefix' for file based cache

### LINKED ISSUE #69465: jsonfile cache plugin prefix option does not work
##### SUMMARY

Ansible documents a fact_caching_prefix option for the jsonfile cache plugin. It does not appear to do anything. Also, the fact_caching_connection option is documented as `_uri` rather than `_connection`.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

jsonfile

##### ANSIBLE VERSION

```paste below
ansible 2.9.8
  config file = /home/mark/src/ansible/ansible2/ansible.cfg
  configured module search path = [u'/home/mark/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/mark/src/ansible-playground/venv-2.9/local/lib/python2.7/site-packages/ansible
  executable location = /home/mark/src/ansible-playground/venv-2.9/bin/ansible
  python version = 2.7.12 (default, Oct  8 2019, 14:14:10) [GCC 5.4.0 20160609]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

Ubuntu

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```paste below

```

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

fixes #69465

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

file based cache plugins

### CODE REVIEW

File: lib/ansible/plugins/cache/__init__.py [hunk 1]
```diff
@@ -137,6 +137,14 @@ def validate_cache_connection(self):
                     raise AnsibleError("error in '%s' cache, configured path (%s) does not have necessary permissions (rwx), disabling plugin" % (
                         self.plugin_name, self._cache_dir))
 
+    def _get_cache_file_name(self, key):
+        prefix = self.get_option('_prefix')
+        if prefix:
+            cachefile = "%s/%s_%s" % (self._cache_dir, prefix, key)
```
Comment (imjoseangel) [inline lib/ansible/plugins/cache/__init__.py]: @bcoca , Personally I prefer the "New Style" String formating. 

```python
cachefile = "{0}/{1}_{2}".format(self._cache_dir, prefix, key)
````

File: lib/ansible/plugins/cache/__init__.py [hunk 2]
```diff
@@ -137,6 +137,14 @@ def validate_cache_connection(self):
...
+        else:
+            cachefile = "%s/%s" % (self._cache_dir, key)
```
Comment (imjoseangel) [inline lib/ansible/plugins/cache/__init__.py:145]: @bcoca , Personally I prefer the "New Style" String formating. 

```python
cachefile = "{0}/{1}".format(self._cache_dir, key)
````

File: lib/ansible/plugins/cache/__init__.py (hunk 1 above)
Comment (markgoddard) [inline lib/ansible/plugins/cache/__init__.py]: Does the addition of an underscore match the behaviour of RDBMS caches?

Comment (imjoseangel) [reply lib/ansible/plugins/cache/__init__.py]: Agree. To me the prefix should be just a prefix. If prefix is `xxx` for instance, it should be:

`xxxlocalhost`

Adding a prefix like `xxx_` will return `xxx_localhost` then

**Comment by samdoran** [comment]
Please add tests. 🙏

**Review by jimi-c — APPROVED** [review]

**Review by s-hertel — APPROVED** [review]

**Comment by markgoddard** [comment]
Thanks!

**Pull request merged**